### PR TITLE
Fix timeouts gain on DC became inaccessible

### PR DESCRIPTION
**Pull request by sabramkin**
Rework of the branch https://github.com/shaitan/elliptics/tree/debug

### CODE REVIEW

File: library/trans.c [hunk 1]
```diff
@@ -442,41 +444,54 @@ void dnet_trans_clean_list(struct list_head *head, int error)
...
-					 dnet_state_dump_addr(st), t->cmd.backend_id,
+					 dnet_state_dump_addr(st), cmd->backend_id,
 					 old_cache_weight, new_cache_weight,
 					 old_disk_weight, new_disk_weight);
 			}
 		}
 	}
 }
 
+int dnet_timespec_less(struct timespec *l, struct timespec *r)
```
Comment (shaitan) [inline library/trans.c]: How about making and using more general variant?
```suggestion
static int dnet_timespec_cmp(struct timespec *l, struct timespec *r)
```

File: library/elliptics.h [hunk 2]
```diff
@@ -136,6 +136,7 @@ struct dnet_net_state
 	int			__need_exit;
 
 	int			stall;
+	struct timespec		stall_inc_ts;
```
Comment (shaitan) [inline library/elliptics.h]: I would call it just `stall_ts`

File: library/trans.c (hunk 1 above)
Comment (sabramkin) [reply library/trans.c]: `static` ok. Are you sure about `_cmp`? Functions with such a postfix usually return values -1, 0, 1 (less, eq, greater).

Comment (sabramkin) [reply library/trans.c]: Ok, I understand, you meant general variant.

**Pull request merged**